Transport refactor

shafreenAnfar opened the pull request:
This transport contains some major changes. Hence, we thought of doing it step by step. 

As a starting point we changed HTTP, WebSockets and JMS connectors. Once this is completed we should be able to to get rid of carbon-messaging.

irunika committed:
Add necessary interfaces for WebSocket message transportation

irunika committed:
Add common interface for all WebSocket carbon messages

irunika committed:
Refactor WebSocketMessageProcessor and add doc comments.

irunika committed:
Refactor WebSocketMessage and add java docs

irunika committed:
Add necessary methods for WebSocketTextMessage interface

irunika committed:
Introduce WebSocket Session interface

irunika committed:
Introduce necessary methods for WebSocketInitMessage

irunika committed:
Introduce necessary methods for WebSocketBinaryMessage

irunika committed:
Introduce necessary methods for WebSocketCloseMessage

irunika committed:
Introduce necessary methods for WebSocketControlMessage

Commit:
Include initial design concept

irunika committed:
Introduce WebSocket message properties manager

Commit:
Remove handleInboudRequest

irunika committed:
Introduce error handler for WebSocket message processor

irunika committed:
Introduce method to check the inbound message source

irunika committed:
Introduce WebSocket client connector

irunika committed:
Change the class name of WebSocketMessagePropertiesManager to WebSocketMessagePropertiesHandler

Commit:
Update the design

irunika committed:
Modify interfaces of WebSocketSessionContext

irunika committed:
Change interfaces to support observer pattern

irunika committed:
Implement WebSocketObservable

irunika committed:
Implement WebSocketObservable

irunika committed:
Introduce context holder for WebSocket

irunika committed:
Implement WebSocket channel handler context

irunika committed:
Implement WebSocketInitMessage

irunika committed:
Implement WebSocketTextMessage

irunika committed:
Implement WebSocketBinaryMessage

irunika committed:
Implement WebSocketControlMessage

irunika committed:
Implement WebSocketCloseMessage

irunika committed:
modify WebSocket Observable and Observer

irunika committed:
Change the implementation of source handlers to support observer pattern

Commit:
Introduce new interfaces

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/SourceHandler.java
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/WebSocketSourceHandler.java

irunika committed:
Merge HTTP changes of Observer pattern

Commit:
Change the contract

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

Commit:
Remove unwanted classes

Commit:
Add HTTPClientConnector interface

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/WebSocketSourceHandler.java

irunika committed:
Refactor code base for new listeners

Commit:
Fix the single thread issue

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

irunika committed:
Modify WebSocket server test cases

Commit:
Fix client connector

irunika committed:
Implement WebSocket client connector and add test cases

Commit:
Change to use HTTPCarbonMessage

irunika committed:
Add handshake with idle timeout

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

irunika committed:
Add constants

Commit:
Fix the client connector

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

shafreenAnfar committed:
Merge pull request #2 from irunika/websocket-messaging-framework

[WebSocket] Transport Refactoring

irunika committed:
Add pass-through support

irunika committed:
Clean the code

Commit:
Fix test cases

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractImpl/Main.java
#	http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/websocket/WebSocketServerTest.java

irunika committed:
Fix ServerBootstrapConfiguration issue

Commit:
Fix content aware testcases

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/channel/ChannelUtils.java

lankavitharana committed:
Refactor jms transport implementation

Commit:
Fix testcases

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

irunika committed:
Fix bugs and naming conventions

irunika committed:
Fix test cases

Commit:
Fix null pointer issue in testcases

Commit:
Remove thread sleep

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/message/HTTPCarbonMessage.java
#	http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/contentaware/RequestResponseCreationListener.java

irunika committed:
Clean the code

Commit:
Remove blocking operation

Commit:
Remove unwanted code lines

shafreenAnfar committed:
Merge pull request #4 from irunika/websocket-messaging-framework

Transport Refatoring

Commit:
Remove unwanted imports

Commit:
Add a new util method

Commit:
Make the buildListner method public

irunika committed:
Remove unnecessary classes and clean code

lankavitharana committed:
Remove unnessary line

shafreenAnfar committed:
Merge pull request #3 from lankavitharana/transRefac

Refactor jms transport implementation

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPClientConnectorFuture.java:
```diff
@@ -0,0 +1,12 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPClientConnector.java:
```diff
@@ -0,0 +1,12 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPConnectorFactory.java:
```diff
@@ -0,0 +1,19 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPConnectorListener.java:
```diff
@@ -0,0 +1,11 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/ServerConnector.java:
```diff
@@ -0,0 +1,9 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/ServerConnectorFuture.java:
```diff
@@ -0,0 +1,83 @@
+package org.wso2.carbon.transport.http.netty.contract;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPClientConnectorFutureImpl.java:
```diff
@@ -0,0 +1,28 @@
+package org.wso2.carbon.transport.http.netty.contractimpl;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPClientConnectorImpl.java:
```diff
@@ -0,0 +1,86 @@
+package org.wso2.carbon.transport.http.netty.contractimpl;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPConnectorFactoryImpl.java:
```diff
@@ -0,0 +1,62 @@
+package org.wso2.carbon.transport.http.netty.contractimpl;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPResponseListener.java:
```diff
@@ -0,0 +1,144 @@
+package org.wso2.carbon.transport.http.netty.contractimpl;
```
Shall we add License header?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPServerConnectorFuture.java:
```diff
@@ -0,0 +1,87 @@
+package org.wso2.carbon.transport.http.netty.contractimpl;
```
Shall we add License header?

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/websocket/message/WebSocketBinaryMessageImpl.java:
```diff
@@ -0,0 +1,66 @@
+    public byte[] getByteArray() {
+        byte[] bytes;
+        if (buffer.hasArray()) {
+            return buffer.array();
```
To be consistent with the flow, make this "bytes = buffer.array()", since a "bytes" variable is declared in the first line, and its returned in the end.

shafreenAnfar replied:
@irunika

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPClientConnector.java:
```diff
@@ -0,0 +1,12 @@
+import org.wso2.carbon.transport.http.netty.message.HTTPCarbonMessage;
+
+/**
+ * Allows to send outbound messages
```
Main comment must end with a full stop ("."), fix in other places as well.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPClientConnector.java:
```diff
@@ -0,0 +1,12 @@
+ * Allows to send outbound messages
+ */
+public interface HTTPClientConnector {
+    HTTPClientConnectorFuture connect();
```
Add Javadoc comments for interface methods.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPClientConnectorFuture.java:
```diff
@@ -0,0 +1,12 @@
+import org.wso2.carbon.transport.http.netty.message.HTTPCarbonMessage;
+
+/**
+ * HTTPClientConnectorFuture
```
Add a meaningful comment, fix in other places also.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/ServerConnectorFuture.java:
```diff
@@ -0,0 +1,83 @@
+    /**
+     * Notify HTTP messages to the listener.
+     *
+     * @param httpMessage HTTP message.
```
Param comments ends with a period only if its a sentence. Check http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#styleguide.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPClientConnectorImpl.java:
```diff
@@ -0,0 +1,86 @@
+        } else {
+            host = Constants.LOCALHOST;
+            httpCarbonMessage.setProperty(Constants.HOST, Constants.LOCALHOST);
+            log.debug("Cannot find property HOST of type string, hence using localhost as the host");
```
debug messages should be wrapped in if (log.isDebugEnabled()) blocks.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPClientConnectorImpl.java:
```diff
@@ -0,0 +1,86 @@
+
+        SourceHandler srcHandler = (SourceHandler) httpCarbonMessage.getProperty(Constants.SRC_HANDLER);
+        if (srcHandler == null) {
+            log.debug("SRC_HANDLER property not found in the message." +
```
Use the constant Constants.SRC_HANDLER for the message as well.

lafernando commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/websocket/message/WebSocketInitMessageImpl.java:
```diff
@@ -0,0 +1,147 @@
+
+    /* Get the URL of the given connection */
+    private String getWebSocketURL(HttpRequest req) {
+        String protocol = "ws";
```
"ws" and "wss" must be defined as constants.

shafreenAnfar replied:
@irunika

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/ServerConnectorException.java:
```diff
@@ -17,26 +17,22 @@
 
-    public WebSocketCallback(CountDownLatch countDownLatch) {
-        this.countDownLatch = countDownLatch;
+    public ServerConnectorException(Exception e) {
```
Please document these public methods for all possible cases.

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/websocket/WebSocketControlSignal.java:
```diff
@@ -0,0 +1,27 @@
+ * Control signals for WebSocket connection.
+ */
+public enum WebSocketControlSignal {
+    PING, PONG;
```
Please document these enum fields, what they represent?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPResponseListener.java:
```diff
@@ -0,0 +1,144 @@
+                ctx.write(httpContent);
+            }
+
+//            if (httpMessage instanceof HTTPCarbonMessage) {
```
May be we can get rid of this commented code.

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HTTPResponseListener.java:
```diff
@@ -0,0 +1,144 @@
+
+    @Override
+    public void onError(Throwable throwable) {
+
```
Do we need to handle any logic here for error scenario on registered listener? Was it intentional to left this as empty?

djkevincr commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/listener/HTTPServerConnectorProvider.java:
```diff
@@ -18,104 +18,77 @@
+//            serverConnectorBootstrap = new ServerConnectorBootstrap(trpConfig);
+//            serverConnectorBootstrap.initialize();
+//        }
+//    }
```
May be we can clean up the commented class.

shafreenAnfar replied:
Removed the class.

irunika committed:
Change the interface name

irunika committed:
Add convert methods for WebSocket

Commit:
Add class level documentation

Commit:
Add java doc to interefaces

Commit:
Add java doc for WebSocketClientConnector

irunika committed:
Add HTTP util method to extract transport properties

Commit:
Add java doc to HTTPClientConnectorFutureImpl

Commit:
Add java doc to HTTPClientConnectorImple

Commit:
Remove unwanted comments

irunika committed:
Add method to get ID for HTTP ServerConnector

shafreenAnfar committed:
Merge pull request #5 from irunika/websocket-messaging-framework

[WebSocket] Listener Interface Changes

irunika committed:
Fix bug

irunika committed:
Add doc comments to WebSocketClientConnector

irunika committed:
Fix bugs in WebSocket client connector

irunika committed:
Change the implementation of WebSocket messages

irunika committed:
Fix issues in test cases

irunika committed:
Clean the code

irunika committed:
Change the method signature of getWSClientConnector

irunika committed:
Clean the test cases code and complete java docs

irunika committed:
Add headers feature for WebSocket message

irunika committed:
Add missing properties for WebSocket init message

irunika committed:
Change the names of the test cases

Commit:
Fix integration testcases

Commit:
Refactor code readability

Commit:
Remove unwanted code lines

irunika committed:
Add new test case to check WebSocket message properties

irunika committed:
Merge remote-tracking branch 'shafreen/transport-refactor' into websocket-messaging-framework

# Conflicts:
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/HTTPConnectorFactory.java
#	http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contract/websocket/WebSocketClientConnector.java

irunika committed:
Add java docs

Commit:
Refactor code to be more readable

shafreenAnfar committed:
Merge pull request #6 from irunika/websocket-messaging-framework

Bug Fixes

Commit:
Fix the null pointer issue

Commit:
Refactor interface names

Commit:
Remove unwanted code lines

irunika committed:
Add TestNG listener to indicate the current test case`

Commit:
Add sync method to ServerConnectorFuture

shafreenAnfar committed:
Merge pull request #7 from irunika/transport-refactor

Transport refactor

pubudu91 committed:
Add HTTP trace logging capability

Commit:
Renamve the function names

Commit:
Add doc comments

pubudu91 committed:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into wire-logs-tr

Commit:
Change the api not throw exception

pubudu91 committed:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into wire-logs-tr

Commit:
Fix test failures

pubudu91 committed:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into wire-logs-tr

pubudu91 committed:
Migrate HTTP trace logging to the transport refactor

irunika committed:
Change the behavior of test case logging

irunika committed:
Add date for the logging

irunika committed:
Remove unnecessary logs from test cases

irunika committed:
Add necessary logs

irunika committed:
Modify logs

irunika committed:
Fix error occurred in WebSocketRemoteServerFrameHandler

irunika committed:
Resolve warnings from check style plugin

irunika committed:
Remove unused packages from pom.xml

irunika committed:
Remove org.wso2.carbon.transport.http.netty.message from internal packages in pom.xml
Since this package is publicly used in APIs now, a build warning is given about it. Because it is publicly used there is no point of treating is as an internal package.

Commit:
Add utile method to get listener interface from parameters

shafreenAnfar committed:
Merge pull request #8 from irunika/transport-refactor

Transport Refactor

Commit:
Fix check-style issue

Commit:
Add error notifiyer

Commit:
Resolve merge conflicts

Commit:
Remove unwanted log entry

irunika committed:
Update version from 4.x.x to 5.0.0-SNAPSHOT

Commit:
Change HTTP to Http in method names

pubudu91 committed:
Refactor server connector trace log enable code

pubudu91 committed:
Replace sender config with a boolean for enabling HTTP trace logs

pubudu91 committed:
Refactor HTTP trace log enabling in server connector bootstrap

shafreenAnfar commented on http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/websocket/WebSocketClientConnectorImpl.java:
```diff
@@ -0,0 +1,70 @@
+                                                              customHeaders, sourceHandler, connectorListener);
+        try {
+            webSocketClient.handshake();
+            return webSocketClient.getSession();
```
Can't we return session upon successful handshake ?

irunika committed:
Modify test cases to sync with the server

pubudu91 committed:
Resolve conflicts

shafreenAnfar committed:
Merge pull request #9 from pubudu91/wire-logs-tr

Add HTTP trace log support

shafreenAnfar commented on jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/contract/JMSListener.java:
```diff
@@ -0,0 +1,12 @@
+ * Allows to get notifications.
+ */
+public interface JMSListener {
+    void onMessage(CarbonMessage jmsMessage, JMSCallback jmsCallback);
```
@lankavitharana Shall we add api doc ?

Commit:
Remove http server connector provider

Commit:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into transport-refactor

lankavitharana committed:
Add javadoc comments

Commit:
Move adding log handler to IO thread

shafreenAnfar committed:
Merge pull request #10 from lankavitharana/transRefac

Add javadoc comments

irunika committed:
Resolve requested changes in PR

Commit:
Update sever boot strap addmethods

irunika committed:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into transport-refactor

irunika committed:
Remove unnecessary ListenerConfiguration classes

shafreenAnfar committed:
Merge pull request #11 from irunika/transport-refactor

Transport Refactoring

irunika committed:
Resolve requested changes in PR

Commit:
Remove uwanted classes

irunika committed:
Merge branch 'transport-refactor' of https://github.com/shafreenAnfar/carbon-transports into transport-refactor

shafreenAnfar committed:
Merge pull request #12 from irunika/transport-refactor

Transport Refactoring

anupama-pathirage commented on jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/contract/JMSClientConnector.java:
```diff
@@ -0,0 +1,23 @@
+package org.wso2.carbon.transport.jms.contract;
```
Add license header. Check all the new classes in org.wso2.carbon.transport.jms package

irunika replied:
Done

Commit:
Remove thread sleeps from testutils

Commit:
Remove unwanted variables

irunika committed:
Fix java doc warnings

irunika committed:
Modify WebSocket error handling

shafreenAnfar committed:
Merge pull request #13 from irunika/transport-refactor

Transport Refactoring

anupama-pathirage commented on http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/server/ServerThread.java:
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
```
Incorrect copyright year

irunika replied:
Corrected.

Commit:
Add flush methods to test backend

lankavitharana committed:
Add license headers

shafreenAnfar committed:
Merge pull request #14 from lankavitharana/transRefac

Add license headers

irunika committed:
Correct warnings of JMS

irunika committed:
Change the wrong license headers

shafreenAnfar committed:
Merge pull request #15 from irunika/transport-refactor

Transport refactor

pubudu91 committed:
Fix Java doc warnings

shafreenAnfar committed:
Merge pull request #16 from pubudu91/transport-refactor

Fix Java doc warnings
